cmd/agent/app/run.go

```diff
@@ -374,7 +374,8 @@ func StartAgent() error {
 	}
 
 	// start dependent services
-	startDependentServices()
+	go startDependentServices()
```

Comment:
what happens if a dependent service attempts to be started here: https://github.com/DataDog/datadog-agent/blob/ed5a804ebf2d275d7397bcc66a36a258fbfcebb6/cmd/agent/app/dependent_services.go#L29-L34 while the main agent service is still reporting itself as `StartPending` (not yet as `Running`)?

Will the dependent service `Start` call fail immediately, or will it wait until the main agent service is `Running`, or will it have some other behavior?

I want to make sure we understand how things behave in this scenario, since this scenario couldn't happen before this PR.

Reply:
it will block (which is why we have to make it async).  Since it blocks, then moving the `started` notification until after `StartAgent()` caused a deadlock, b/c the synchronous `startdependentservices()` was blocked.

Reply:
ok, I think this covers the concern I had then, thanks.